src/category_theory/over.lean

```diff
@@ -130,6 +131,41 @@ instance forget_reflects_iso : reflects_isomorphisms (forget X) :=
 { reflects := λ Y Z f t, by exactI
   { inv := over.hom_mk t.inv ((as_iso ((forget X).map f)).inv_comp_eq.2 (over.w f).symm) } }
 
+instance forget_faithful : faithful (forget X) := {}.
+
+/--
+If `k.left` is an epimorphism, then `k` is an epimorphism. In other words, `over.forget X` reflects
+epimorphisms.
+The converse does not hold without additional assumptions on the underlying category.
+-/
+-- TODO: Show the converse holds if `T` has binary products or pushouts.
+lemma epi_of_epi_left {f g : over X} (k : f ⟶ g) [hk : epi k.left] : epi k :=
+faithful_reflects_epi (forget X) hk
+
+/--
+If `k.left` is a monomorphism, then `k` is a monomorphism. In other words, `over.forget X` reflects
+monomorphisms.
+The converse of `category_theory.over.mono_left_of_mono`.
```

Comment:
```suggestion
The converse of `category_theory.over.mono_left_of_mono`.

This lemma is not an instance, to avoid loops in type class inference.
```